File: Python/0001_two_sum/solution.py
def twoSum(nums, target):
    num_map = {}
    for index, num in enumerate(nums):
        complement = target - num
        if complement in num_map:
            return [num_map[complement], index]
        num_map[num] = index  
    return [] 

nums = [2,7,11,15]
target = 9
print(twoSum(nums,target))

File: JavaScript/0001_two_sum/solution.js
function two_sum(nums,target) {
    let num_map = {};
    for (let [index,num] of nums.entries()) {
        let complement = target - num;
        if (complement in num_map) {
            return [num_map[complement], index];
        }
        num_map[num] = index;
    }
    return [];
}

const nums = [2,7,11,15];
const target = 9;
console.log(two_sum(nums,target));